src/functions/indAffine.jl

```diff
@@ -26,11 +26,11 @@ immutable IndAffine{T <: RealOrComplex} <: IndicatorConvex
   end
 end
 
-IndAffine{T <: RealOrComplex}(A::AbstractArray{T,2}, b::AbstractArray{T,1}) =
-  IndAffine{T}(A, b)
+IndAffine{T <: RealOrComplex, M<:AbstractArray{T,2}, V<:AbstractArray{T,1}}(A::M, b::V) =
+  IndAffine{T,M,V}(A, b)
```

Comment:
I see, the problem with inner constructors is that we need to know all the parameters in the type when the constructor is called. I'm not sure how this is possible, for example `F` in the case with `LeastSquares` if the factorization is done in the inner constructor.